src/workerd/api/hibernatable-web-socket.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <kj/debug.h>
+
+#include <workerd/io/worker-interface.capnp.h>
+#include <workerd/api/global-scope.h>
+#include <workerd/io/worker-interface.h>
+
+namespace workerd::api {
+
+class HibernatableWebSocketEvent final: public ExtendableEvent {
+public:
+  explicit HibernatableWebSocketEvent();
+
+  static jsg::Ref<HibernatableWebSocketEvent> constructor(kj::String type) = delete;
+
+  // TODO(soon): return correct ws instead of the current stub implementation
+  jsg::Ref<WebSocket> getWebSocket(jsg::Lock& lock);
+  jsg::Value getError(jsg::Lock& lock);
+
+  JSG_RESOURCE_TYPE(HibernatableWebSocketEvent) {
+    JSG_INHERIT(ExtendableEvent);
+  }
+};
+
+struct HibernatableSocketParams {
+  enum Type {
+    TEXT,
+    DATA,
+    CLOSE,
+    ERROR
+  };
+
+  Type type;
+  kj::Array<byte> data;
+  kj::String message;
+  kj::String closeReason;
+  int closeCode;
```

Comment:
Just realized we should also have a `wasClean` property on close events: https://github.com/cloudflare/workerd/blob/e8c02bef30c96e60181c656ad2052a2e762a0d1a/src/workerd/api/web-socket.h#L94

See: https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent/wasClean

I didn't even know this was a thing, only noticed because I'm working on disconnects now. @jqmmes is it alright with you if I add that property to `HibernatableSocketParams` and include it as a parameter in the close event as part of my HibernationManager PR?

Reply:
Sure!

Reply:
Thanks, have a great weekend!